### DISCUSSION TITLE
Internal Metrics - Emit only required parameters

### DISCUSSION THREAD

**Discussion by hemanthofficial3009** [vectordotdev/vector#12114] [Q&A] [answered]
Hello Team, 
Is there a way to configure the Internal Metrics (for _vector_ instance acting as client which is configured to received the events from http component), so that it would emit only the required parameters such as _component_sent_events_total_ , _component_received_events_total_ etc., rather than having all the parameter flushed out to the sink file.?

**Comment by spencergilbert** [comment, marked as answer]
Hi @hemanthofficial3009 - that's not possible in the source's configuration today, but you could use the `filter` transform to select only the metrics you want to eventually emit.

**Reply by hemanthofficial3009** [reply]
@spencergilbert thanks for the response.
